nixos/modules/programs/ssh.nix

```diff
@@ -204,11 +204,12 @@ in
         serviceConfig =
           { ExecStartPre = "${pkgs.coreutils}/bin/rm -f %t/ssh-agent";
             ExecStart =
-                "${cfg.package}/bin/ssh-agent " +
+                "${cfg.package}/bin/ssh-agent -D " +
                 optionalString (cfg.agentTimeout != null) ("-t ${cfg.agentTimeout} ") +
                 "-a %t/ssh-agent";
+            ExecStartPost = "/bin/sh -c 'set -eu ; while [ ! -S $SSH_AUCK_SOCK ]; do sleep 1; done'";
```

Comment:
I wonder, isn't sleeping for a whole second between tests a bit excessive? `sleep` accepts fractional arguments.